File: 0125-valid-palindrome/0125-valid-palindrome.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        s = s.lower().strip()
        
        # Remove non-alphanumeric characters and spaces.
        cleanedText = [char for char in s if char.isalnum()]
        cleanedText = "".join(cleanedText)

        # If empty string, return True.
        if not cleanedText:
            return True
        
        # Using two pointers, check left and right until they meet in the middle.
        lPtr = 0
        for rPtr in range(len(cleanedText) - 1, -1, -1):
            if lPtr >= rPtr:
                return True
            
            if cleanedText[lPtr] == cleanedText[rPtr]:
                lPtr += 1
            else:
                return False
                

File: 0125-valid-palindrome/0125-valid-palindrome.java
class Solution {
    public boolean isPalindrome(String s) {
        String lowerString = s.toLowerCase();
        String finalString = "";
        
        // Get only alphanumeric characters.
        for (int i = 0; i < lowerString.length(); i++) {
            char currChar = lowerString.charAt(i);
            if (Character.isAlphabetic(currChar) || Character.isDigit(currChar)) {
                finalString += currChar;
            }
        }
        
        if (finalString.isEmpty()) {
            return true;
        }
        
        // Use two pointers and see if they meet.
        int rPtr = finalString.length() - 1;
        for (int lPtr = 0; lPtr < finalString.length(); lPtr++) {
            if (lPtr >= rPtr) {
                return true;
            }
            
            if (finalString.charAt(lPtr) != finalString.charAt(rPtr)) {
                return false;
            } else {
                rPtr -= 1;
            }
        }
        return true;
    }
}